src/java.base/share/native/libnet/InetAddress.c

```diff
@@ -81,3 +81,17 @@ JNIEXPORT jboolean JNICALL
 Java_java_net_InetAddress_isIPv4Available(JNIEnv *env, jclass clazz) {
     return ipv4_available();
 }
+
+/*
+ * Class:     java_net_InetAddress
+ * Method:    isIPv6Supported
+ * Signature: ()I
+ */
+JNIEXPORT jboolean JNICALL
+Java_java_net_InetAddress_isIPv6Supported(JNIEnv *env, jclass clazz) {
+    if (ipv6_available()) {
+        return JNI_TRUE;
+    } else {
+        return JNI_FALSE;
+    }
```

Comment:
I don't have knowledge of C or JNI, but the `Java_java_net_InetAddress_isIPv4Available` currently does:

```
return ipv4_available();
```
So maybe a similar construct can be used here instead of if/else blocks? I'm guessing `ipv6_available()` will be returning values that are compatible with `JNI_TRUE/JNI_FALSE`.

Reply:
I agree that it could be simplified to match `Java_java_net_InetAddress_isIPv4Available`. Changed in 49fdd576cade2e97639f827f9db6d0f1e31101e2

Reply:
Thanks for noticing that Jaikiran!

Both `ipv4_available()` and `ipv6_available()` are defined to return `jint` so the implementation in `Java_java_net_InetAddress_isIPv6Supported` is arguably the more correct (provided that the method being called returns 0 when the corresponding IP version is not available).

The signature in the comment should be changed as you suggest though.

Possibly we should consider having both methods use the same construct...

Reply:
Current version should be ok since variables returned by `ipv4_available()` and `ipv6_available()` can only be initialized to `JNI_FALSE` or `JNI_TRUE`:
- `Java_java_net_InetAddress_isIPv4Available`[InetAddress.c] returns result of `ipv4_available`[net_util.c]
- `ipv4_available`[net_util.c] returns `IPv4_available` variable value
- `IPv4_available` value is initalized with a result returned by `IPv4_supported()`[net_util_md.c]
- `IPv4_supported()`[net_util_md.c] can return `JNI_TRUE` or `JNI_FALSE` only (in both Unix and Windows implementations)
-  Same stands true for `ipv6_available()`: only `JNI_TRUE` or `JNI_FALSE` can be returned

Reply:
Thank you Aleksei for those details and the changes.